### ISSUE TITLE
mysql-parser discard comments

### ISSUE THREAD

**Issue by asmgit** [VeryCrazyDog/mysql-parser#1]
Can mysql-parser just split statement without discarding comments? Comments may be needed if an error occurs and query identification.

**Comment by VeryCrazyDog** [comment]
Not the current v1.1.0, however I am open for suggestion to make the module better.

In the current version, if an error occurred, you can log the SQL which trigger the error and search in the SQL before splitting to look for the comment.

If the comments are retained, then you can log the SQL together with comments, and then you can view both comments and SQL in the log, is that how retaining comment helps?

But one question, what is your expectation on the split output of the following SQL?
```sql
-- Comment 1
-- Comment 2
SELECT 'Hello world!' message FROM dual;
SELECT 'Bye world!' message FROM dual;
```

**Comment by asmgit** [comment]
I just want see queries in query_log exactly as I wrote them. Foe example, 2 identical queries with different comment will make me understand from witch module its running, maybe first query running seldom, as expected, but second run very offten.

> But one question, what is your expectation on the split output of the following SQL?
I expect
1.
```sql
-- Comment 1
-- Comment 2
SELECT 'Hello world!' message FROM dual
```
2.
```sql
SELECT 'Bye world!' message FROM dual
```

For next code
```sql
-- Comment 1
-- Comment 2
SELECT 'Hello world!' message FROM dual /*multicomment*/; -- Comment 3
-- Comment 4
SELECT 'Bye world!' message FROM dual
-- Comment 5
;



```
I expect
1.
```sql
-- Comment 1
-- Comment 2
SELECT 'Hello world!' message FROM dual /*multicomment*/
```
2.
```sql
-- Comment 3
-- Comment 4
SELECT 'Bye world!' message FROM dual
-- Comment 5
```

For next code:
```sql
SELECT 'Hello world!' message FROM dual; -- Comment 3
```
1.
```sql
SELECT 'Hello world!' message FROM dual
```
2.
```sql
-- Comment 3
```

Ideal algorithm:
1. split without comment removing
2. trim (\n \t <_space_>) each query
3. remove empty queries

**Comment by asmgit** [comment]
Hmmm...
I checked examples above in query_log for mysql (cli mysql), in fact it cuts out all comments before writing to query_log. Check in PhpStorm, it trim all before and all after query comments, left only inside comments

**Comment by VeryCrazyDog** [comment]
Well, I think it is reasonable that the comments are removed in `query_log` of MySQL. In that case, you can only rely on the application side (your Node.js app) to log the query executed with comments along with it.

Do you think it is still helpful if this *mysql-parser* retain your comments in split result in a way that is the same as your expected output?

**Comment by asmgit** [comment]
Yes, I think it will still helpful, but it is not very critical.
3 mode discarding comments:
- discard all - default - already realised
- before and after - as PHPStorm
- no discard - same as in my examples

I think no reason make sql logging in my Node app, it just double work.
It’s better to make efforts for convenient use mysql log, for example, put a comment label on all queries on application session that you want to trace and have convenient UI for viewing this queries.

**Comment by VeryCrazyDog** [comment]
Guess I will just implement *no discard* in this case. Probably the *before and after* implementation is PHPStorm specific which I don't want to tailor made a feature for only 1 single IDE.

**Comment by VeryCrazyDog** [comment]
Implemented and published in v1.2.0, closing.
